packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -145,6 +170,131 @@ export class RichTextEditor extends FoundationElement {
+            this.tiptapEditor.commands.focus();
+        }
+    }
+
+    /**
+     * This function returns markdown string by serializing tiptap editor document using prosemirror MarkdownSerializer
+     * @public
+     */
+    public getMarkdown(): string {
+        if (this.$fastController.isConnected) {
+            const markdownContent = this.markdownSerializer.serialize(
+                this.tiptapEditor.state.doc
+            );
+            return markdownContent;
+        }
+        return '';
+    }
+
+    /**
+     * This function takes the Fragment from parseMarkdownToDOM function and return the serialized string using XMLSerializer
+     */
+    private getHtmlContent(value: string): string {
+        const markdownDocumentFragment = this.parseMarkdownToDOM(value);
+        return this.xmlSerializer.serializeToString(markdownDocumentFragment);
+    }
+
+    private initializeMarkdownParser(): MarkdownParser {
+        /**
+         * It configures the tokenizer of the default Markdown parser with the 'zero' preset.
+         * The 'zero' preset is a configuration with no rules enabled by default to selectively enable specific rules.
+         * https://github.com/markdown-it/markdown-it/blob/2b6cac25823af011ff3bc7628bc9b06e483c5a08/lib/presets/zero.js#L1
+         *
+         */
+        const zeroTokenizerConfiguration = defaultMarkdownParser.tokenizer.configure('zero');
+
+        // The detailed information of the supported rules were provided in the below CommonMark spec document.
+        // https://spec.commonmark.org/0.30/
+        const supportedTokenizerRules = zeroTokenizerConfiguration.enable([
+            'emphasis',
+            'list'
+        ]);
+
+        return new MarkdownParser(
+            schema,
+            supportedTokenizerRules,
+            defaultMarkdownParser.tokens
+        );
+    }
+
+    private initializeMarkdownSerializer(): MarkdownSerializer {
+        /**
+         * orderedList Node is getting 'order' attribute which it is not present in the
+         * tip-tap orderedList Node and having start instead of order, Changed it to start (nodes.attrs.start)
+         * Assigned updated node in place of orderedList node from defaultMarkdownSerializer
+         * https://github.com/ProseMirror/prosemirror-markdown/blob/b7c1fd2fb74c7564bfe5428c7c8141ded7ebdd9f/src/to_markdown.ts#L94C2-L101C7
+         */
+        const orderedListNode = function orderedList(
+            state: MarkdownSerializerState,
+            node: Node
+        ): void {
+            const start = (node.attrs.start as number) || 1;
+            const maxW = String(start + node.childCount - 1).length;
+            const space = state.repeat(' ', maxW + 2);
+            state.renderList(node, space, i => {
+                const nStr = String(start + i);
+                return `${state.repeat(' ', maxW - nStr.length) + nStr}. `;
+            });
+        };
```

Comment:
Yes `ol`  has only  `start` attribute, For parsing and serializing we are using `defaultMarkdownParser`  and `defaultMarkdownSerializer` from the `prosemirror-markdown` package, both are using same [schema ](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/schema.ts)internally, Especially In that schema it is having `order` [attribute ](https://github.com/ProseMirror/prosemirror-markdown/blob/b7c1fd2fb74c7564bfe5428c7c8141ded7ebdd9f/src/from_markdown.ts#L251C3-L251C40)for `orderedList` node, But once we set the parsed content to tip-tap editor, internally editor will build its own schema, marks and nodes based on extensions, which don't have order attribute in `orderedList` node and only having start attribute, So taking the `orderedList` [node ](https://github.com/ProseMirror/prosemirror-markdown/blob/b7c1fd2fb74c7564bfe5428c7c8141ded7ebdd9f/src/to_markdown.ts#L94C2-L101C7) from `prosemirror-markdown` package and changed it to `start` instead of `order` is working as expected.

Reply:
To summarize, We are using a combination of `prose-mirror` and `Tiptap` where `TipTap` leverages extension support which will be useful in `@mention` in the future, and also it's internal handling in many cases whereas `prose-mirror` leverages support for markdown serializing and parsing, as Tip Tap doesn't [support markdown](https://tiptap.dev/guide/output#not-an-option-markdown) as output.